cmd/plugin/builder/inventory_plugin.go

```diff
@@ -0,0 +1,142 @@
+	pluginInsertCmd.Flags().StringVarP(&ipiFlags.Vendor, "vendor", "", "", "name of the vendor")
+	pluginInsertCmd.Flags().StringVarP(&ipiFlags.Publisher, "publisher", "", "", "name of the publisher")
+	pluginInsertCmd.Flags().BoolVarP(&ipiFlags.DeactivatePlugins, "deactivate", "", false, "mark plugins as deactivated")
+
+	_ = pluginInsertCmd.MarkFlagRequired("repository")
+	_ = pluginInsertCmd.MarkFlagRequired("vendor")
+	_ = pluginInsertCmd.MarkFlagRequired("publisher")
+	_ = pluginInsertCmd.MarkFlagRequired("manifest")
+
+	return pluginInsertCmd
+}
+
+type inventoryPluginActivateDeactivateFlags struct {
+	Repository        string
+	InventoryImageTag string
+	ManifestFile      string
+	Publisher         string
+	Vendor            string
+}
+
+func newInventoryPluginActivateCmd() *cobra.Command {
+	pluginDeactivateCmd, flags := getActivateDeactivateBaseCmd()
+	pluginDeactivateCmd.Use = "activate"
+	pluginDeactivateCmd.Short = "Activate the existing plugin in the inventory database available on the remote repository"
+	pluginDeactivateCmd.Example = ""
+	pluginDeactivateCmd.RunE = func(cmd *cobra.Command, args []string) error {
+		piOptions := inventory.InventoryPluginUpdateOptions{
+			Repository:        flags.Repository,
+			InventoryImageTag: flags.InventoryImageTag,
+			ManifestFile:      flags.ManifestFile,
+			Vendor:            flags.Vendor,
+			Publisher:         flags.Publisher,
+			DeactivatePlugins: false,
+			ImgpkgOptions:     imgpkg.NewImgpkgCLIWrapper(),
+		}
+		return piOptions.UpdatePluginActivationState()
+	}
+	return pluginDeactivateCmd
+}
+
+func newInventoryPluginDeactivateCmd() *cobra.Command {
+	pluginDeactivateCmd, flags := getActivateDeactivateBaseCmd()
+	pluginDeactivateCmd.Use = "deactivate"
+	pluginDeactivateCmd.Short = "Deactivate the existing plugin in the inventory database available on the remote repository"
+	pluginDeactivateCmd.Example = ""
+	pluginDeactivateCmd.RunE = func(cmd *cobra.Command, args []string) error {
+		piOptions := inventory.InventoryPluginUpdateOptions{
+			Repository:        flags.Repository,
+			InventoryImageTag: flags.InventoryImageTag,
+			ManifestFile:      flags.ManifestFile,
+			Vendor:            flags.Vendor,
+			Publisher:         flags.Publisher,
+			DeactivatePlugins: true,
+			ImgpkgOptions:     imgpkg.NewImgpkgCLIWrapper(),
+		}
+		return piOptions.UpdatePluginActivationState()
+	}
+	return pluginDeactivateCmd
+}
+
+func getActivateDeactivateBaseCmd() (*cobra.Command, *inventoryPluginActivateDeactivateFlags) {
+	var flags = &inventoryPluginActivateDeactivateFlags{}
+
+	var activateDeactivateCmd = &cobra.Command{}
+
+	activateDeactivateCmd.Flags().StringVarP(&flags.Repository, "repository", "", "", "repository to publish plugin inventory image")
+	activateDeactivateCmd.Flags().StringVarP(&flags.InventoryImageTag, "plugin-inventory-image-tag", "", "latest", "tag to which plugin inventory image needs to be published")
```

Comment:
question: This tag would be used mostly for testing (by configuring CLI to use this tag as a discovery source) and for production, it would be always `latest` as stated in the default, correct?

Reply:
That is correct, this is mainly for testing and publishing stuff to a different location if needed. 
For production, we are still assessing the use of `latest` as there might be some restrictions on overriding tags on the production repository.